### PR TITLE
llvm 17 fixes

### PR DESCRIPTION
**Pull request by 2over12**

### CODE REVIEW

File: lib/BC/Optimizer.cpp [hunk 1]
```diff
@@ -46,81 +53,39 @@ namespace remill {
...
-  builder.populateFunctionPassManager(func_manager);
-  builder.populateModulePassManager(module_manager);
-  func_manager.doInitialization();
-  llvm::Function *func = nullptr;
-  while (nullptr != (func = generator())) {
-    func_manager.run(*func);
-  }
-  func_manager.doFinalization();
-  module_manager.run(*module);
+  OptimizeBareModule(module, guide);
```
Comment (2over12) [inline lib/BC/Optimizer.cpp:56]: Not sure what we want to do about this i dont see an easy way to do single function inlining anymore i guess we could probably manually call llvm::InlineFunction etc but that seems excessive for something only called in the trace lifter idk

File: scripts/build.sh [hunk 2]
```diff
@@ -23,11 +23,11 @@ DOWNLOAD_DIR="$( cd "$( dirname "${SRC_DIR}" )" && pwd )/lifting-bits-downloads"
 CURR_DIR=$( pwd )
 BUILD_DIR="${CURR_DIR}/remill-build"
 INSTALL_DIR=/usr/local
-LLVM_VERSION=llvm-16
+LLVM_VERSION=llvm-17
 OS_VERSION=
 ARCH_VERSION=
 BUILD_FLAGS=
-CXX_COMMON_VERSION="0.3.1"
+CXX_COMMON_VERSION="0.6.0"
```
Comment (Ninja3047) [inline scripts/build.sh:30]:
```suggestion
CXX_COMMON_VERSION="0.6.1"
```

File: lib/BC/Util.cpp [hunk 3]
```diff
@@ -1205,17 +1205,6 @@ MoveConstantIntoModule(llvm::Constant *c, llvm::Module *dest_module,
         moved_c = ret;
         return ret;
       }
-      case llvm::Instruction::Select: {
```
Comment (Ninja3047) [inline lib/BC/Util.cpp:1208]: why are we removing this case?

**Review by Ninja3047 — APPROVED** [review]
build stuff looks good
just a question about why the select case is removed

File: lib/BC/Util.cpp (hunk 3 above)
Comment (2over12) [reply lib/BC/Util.cpp:1208]: llvm 17 weakened constant expressions significantly, you cant create a select anymore

**Pull request merged**